ambry-clustermap/src/main/java/com.github.ambry.clustermap/AmbryDatanode.java

```diff
@@ -140,8 +140,7 @@ public int compareTo(DataNodeId o) {
       throw new IllegalStateException("Incompatible objects to compare");
     }
     AmbryDataNode other = (AmbryDataNode) o;
-    int compare = (plainTextPort.getPort() < other.plainTextPort.getPort()) ? -1
-        : ((plainTextPort.getPort() == other.plainTextPort.getPort()) ? 0 : 1);
+    int compare = Integer.compare(plainTextPort.getPort(), other.plainTextPort.getPort());
```

Comment:
Java 8 has some constructs for these multi-field comparison pattern that we often encounter. Here, you could do the following:
```java
return Comparator.comparingInt((AmbryDataNode k) -> k.plainTextPort.getPort()).
                  thenComparing(k -> k.hostName).compare(this, other);
```

Reply:
Thanks, looks more concise.